kernel/kernel-api/src/main/java/io/delta/kernel/internal/CommitRangeFactory.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal;
+
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+import static io.delta.kernel.internal.util.Utils.resolvePath;
+
+import io.delta.kernel.Snapshot;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.internal.fs.Path;
+import io.delta.kernel.internal.lang.ListUtils;
+import io.delta.kernel.internal.util.FileNames;
+import io.delta.kernel.utils.FileStatus;
+import java.util.List;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class CommitRangeFactory {
+
+  private static final Logger logger = LoggerFactory.getLogger(CommitRangeFactory.class);
+
+  private final CommitRangeBuilderImpl.Context ctx;
+  private final Path tablePath;
+  private final Path logPath;
+
+  CommitRangeFactory(Engine engine, CommitRangeBuilderImpl.Context ctx) {
+    this.ctx = ctx;
+    this.tablePath = new Path(resolvePath(engine, ctx.unresolvedPath));
+    this.logPath = new Path(tablePath, "_delta_log");
+  }
+
+  CommitRangeImpl create(Engine engine) {
+    // Resolve startVersion and try to resolve endVersion if possible
+    long startVersion = resolveStartVersion(engine);
+    Optional<Long> endVersionOpt = tryResolveEndVersion(engine);
+    // If we have resolved endVersion, validate that it is >= startVersion
+    endVersionOpt.ifPresent(
+        endVersion ->
+            checkArgument(
+                endVersion >= startVersion,
+                String.format(
+                    "Resolved startVersion=%d > endVersion=%d", startVersion, endVersion)));
+    logger.info(
+        "{}: Resolved startVersion={} and endVersion={} from startBoundary={} endBoundary={}",
+        tablePath,
+        startVersion,
+        endVersionOpt,
+        ctx.startBoundaryOpt,
+        ctx.endBoundaryOpt);
+    // TODO: for now we just store a list of commit files, this will be updated when we add support
+    //  for ccv2
+    List<FileStatus> deltaFiles = getDeltaFiles(engine, startVersion, endVersionOpt);
```

Comment:
I would prefer if we started using the ParsedLogData primitive :/ wdyt?

Reply:
I'd rather do this in a follow-up PR. I think we'll be changing this code anyways for the ccv2 support so let's wait and do it then. For now -- we just super simply and minimally reuse the current getChanges code.